### PR TITLE
fix: allow suspended work to get deleted

### PR DESCRIPTION
**Pull request by a7i**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Based on discussion in #5354 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks @a7i 
/assign
The CI lint is failed.

**Comment by a7i** [comment]
> Thanks @a7i /assign The CI lint is failed.

strange, passes locally.

File: pkg/controllers/execution/execution_controller_test.go [hunk 1]
```diff
@@ -21,12 +21,17 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/objectwatcher"
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/execution/execution_controller_test.go:25]: These two lines may need to go in the third module.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot
/lgtm
/assign @RainbowMango

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**